src/WebJobs.Script.Grpc/MessageExtensions/GrpcMessageConversionExtensions.cs

```diff
@@ -24,93 +24,35 @@ internal static class GrpcMessageConversionExtensions
+            typedData.DataCase switch
+            {
+                    RpcDataType.None => null,
+                    RpcDataType.String => typedData.String,
+                    RpcDataType.Json => JsonConvert.DeserializeObject(typedData.Json, _datetimeSerializerSettings),
+                    RpcDataType.Bytes or RpcDataType.Stream => typedData.Bytes.ToByteArray(),
+                    RpcDataType.Http => GrpcMessageExtensionUtilities.ConvertFromHttpMessageToExpando(typedData.Http),
+                    RpcDataType.Int => typedData.Int,
+                    RpcDataType.Double => typedData.Double,
                     // TODO better exception
-                    throw new InvalidOperationException("Unknown RpcDataType");
-            }
-        }
-
-        public static async Task<TypedData> ToRpc(this object value, ILogger logger, GrpcCapabilities capabilities)
-        {
-            TypedData typedData = new TypedData();
-            if (value == null)
-            {
-                return typedData;
-            }
-
-            if (value is byte[] arr)
-            {
-                typedData.Bytes = ByteString.CopyFrom(arr);
-            }
-            else if (value is JObject jobj)
-            {
-                typedData.Json = jobj.ToString(Formatting.None);
-            }
-            else if (value is string str)
-            {
-                typedData.String = str;
-            }
-            else if (value.GetType().IsArray && IsTypedDataCollectionSupported(capabilities))
-            {
-                typedData = value.ToRpcCollection();
-            }
-            else if (value is HttpRequest request)
-            {
-                typedData = await request.ToRpcHttp(logger, capabilities);
-            }
-            else
-            {
-                typedData = value.ToRpcDefault();
-            }
-
-            return typedData;
-        }
-
-        internal static TypedData ToRpcCollection(this object value)
-        {
-            TypedData typedData;
-            if (value is byte[][] arrBytes)
-            {
-                typedData = arrBytes.ToRpcByteArray();
-            }
-            else if (value is string[] arrStr)
-            {
-                typedData = arrStr.ToRpcStringArray();
-            }
-            else if (value is double[] arrDouble)
-            {
-                typedData = arrDouble.ToRpcDoubleArray();
-            }
-            else if (value is long[] arrLong)
-            {
-                typedData = arrLong.ToRpcLongArray();
-            }
-            else
-            {
-                typedData = value.ToRpcDefault();
-            }
+                    _ => throw new InvalidOperationException($"Unknown RpcDataType: {typedData.DataCase}")
+            };
 
-            return typedData;
-        }
+        public static async Task<TypedData> ToRpc(this object value, ILogger logger, GrpcCapabilities capabilities) =>
```

Comment:
this isn't part of your change, but this seems like a prime candidate to get rid of `async` here and return `Tasks` directly for perf.

Reply:
Taking note to address in another PR